src/web/core/src/main/java/org/geoserver/web/ServiceInfoCapabilitiesProvider.java

```diff
@@ -67,34 +79,13 @@ public Component getCapabilitiesComponent(final String id) {
     /**
      * Check which services are covered by {@link ServiceDescriptionProvider} and can be safely skipped.
      *
-     * <p>This is slightly more error-prone when global services are disabled and few ServiceDescriptionProvider's are
-     * willing to participate. In this case we check each workspace to see what services are available for the
-     * application as a whole.
-     *
      * @return list of services to skip, in lowercase (for case-insensitive matching)
      */
     protected Set<String> skipServiceDescriptionProviders() {
-        GeoServer geoServer = GeoServerApplication.get().getGeoServer();
-
         Set<String> skip = new HashSet<>();
-        List<WorkspaceInfo> workspaceList = new ArrayList<>();
```

Comment:
Whoa, what happened here... I see the service descriptor provider had the service types buried in the ServiceDescriptor and and ServiceLinkDescription it was returning, and you basically exposed them in a new method, instead of having them be hunted down piece by piece.

However how does that relate to the exception you were fixing here? I guess it was happening in the code that got removed. If so, I'd have a test for ServiceInfoCapabilitiesProvider replicating the context of the exception and making sure it's not happening any more.
The kind of test that's easier to write against the un-modified code first.